napalm_yang/parsers/jsonp.py

```diff
@@ -127,6 +127,11 @@ def _parse_container_default(self, attribute, mapping, data):
             return "", {}
         elif "gate" in mapping:
             return None, {}
+        elif "config_default" in mapping:
+            key = mapping.get('key', 'config_default')
+            config_default = self.resolve_path(
+                data[0], mapping["config_default"], mapping.get("default"))
+            return "", {key: config_default['#text']}
```

Comment:
Looks quite good and it's similar to what I was thinking myself. Just two differences to what I was thinking myself. From an API I'd do:

```
- path: "aaa.authorization.policy.local.default-role"
  save: "default_role"
```

And then your implementation would look more like:

```
 elif "save" in mapping:
    return None, {mapping["save"]: d}
```

I also think we should return `None` here. That'd mean that the parser will keep processing rules (not sure if it will save the `extra_vars` though but we can fix that if it doesn't). In that case you could save multiple defaults in the same object or even save defaults right in the same leaf as it's needed. For instance:

```
my_container:
  _process:
    - path: blah.blah
      save: key1
    - path: bleh.bleh
      save: key2
  my_leaf:
    - pre: "{{ etra_vars.key1 }}-{{ extra_vars.key2 }}"
```

or

```
my_container:
  my_leaf:
    - path: blah.blah
      save: key1
    - path: bleh.bleh
      save: key2
    - pre: "{{ etra_vars.key1 }}-{{ extra_vars.key2 }}"
```

Reply:
I couldn't get it to work exactly the way you have it above, and please let me know if im just going further down the wrong road.  But if I return `None` the whole thing stops processing and I don't see how it would continue to process the other save directives.  So for now I left it returning `""` but I added something to the `parsers.base.parse_container` that would allow you to set multiple extra_vars, so we can do what you have posted about.  My latest commit I'm able to grab another part of the config and I tacked it on to the end of the role just as a demo.

Now I haven't found a good example for testing the save directive in a leaf yet, since in the username role example we have here the data available to it are things under the path `username.?name.privilege` . But I'll test the some more later.